### PR TITLE
[gradle-plugin] Support environment variables

### PR DESCRIPTION
**Pull request by ilmat192**
DEBUGGING_SYMBOLS - if 'YES' all artifacts will be built
    with -g flag.

CONFIGURATION_BUILD_DIR - if set all artifacts will be
    placed in the directory specified by this
    environment variable

### CODE REVIEW

File: tools/kotlin-native-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/EnvironmentVariables.kt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package org.jetbrains.kotlin.gradle.plugin
```
Comment (olonho) [inline tools/kotlin-native-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/EnvironmentVariables.kt]: Copyright?

File: tools/kotlin-native-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/KonanBuildingConfig.kt [hunk 2]
```diff
@@ -88,9 +88,28 @@ abstract class KonanBuildingConfig<T: KonanBuildingTask>(private val name_: Stri
         targetToTask[toAdd.konanTarget] = toAdd
     }
 
+    data class OutputPlacement(val destinationDir: File, val artifactName: String)
+
+    // There are two options for output placement.
+    //      1. Gradle's build directory. We use it by default, e.g. if user runs gradle from command line.
```
Comment (olonho) [inline tools/kotlin-native-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/KonanBuildingConfig.kt]: gradle -> Gradle

File: tools/kotlin-native-gradle-plugin/src/test/groovy/org/jetbrains/kotlin/gradle/plugin/test/EnvVariableSpecification.groovy [hunk 3]
```diff
@@ -0,0 +1,217 @@
+package org.jetbrains.kotlin.gradle.plugin.test
```
Comment (olonho) [inline tools/kotlin-native-gradle-plugin/src/test/groovy/org/jetbrains/kotlin/gradle/plugin/test/EnvVariableSpecification.groovy]: Copyright.

File: samples/gtk/build.gradle [hunk 4]
```diff
@@ -1,6 +1,9 @@
 apply plugin: 'konan'
 
-konan.targets = ['macbook', 'linux']
+konan {
+    targets = ['macbook', 'linux']
+    jvmArgs = ['-Xmx8g']
```
Comment (olonho) [inline samples/gtk/build.gradle]: Why so big?

Comment (ilmat192) [reply samples/gtk/build.gradle]: We have such value in `build.sh`.

**Pull request merged**